### PR TITLE
Aggregate CRDs from remote repos

### PR DESCRIPTION
**Pull request by tfussell**
Towards https://github.com/giantswarm/giantswarm/issues/19586

Before we can move APIs out of apiextensions, we need a method to aggregate CRDs from other repos into our CRD "helm charts". This PR adds the ability to list CRDs in a remote GitHub repository in a certain directory (usually `config/crd`). As APIs are moved out, they would be added to `remoteRepositories`.

## Checklist

- [x] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: pkg/crd/renderer.go [hunk 1]
```diff
@@ -24,29 +28,63 @@ var (
...
+	patchedCRDs := make([]runtime.Object, 0, len(crds))
+	for _, crd := range crds {
+		if crdV1, ok := crd.(*v1.CustomResourceDefinition); ok {
+			var err error
+			crd, err = patchCRD(r.Patches, crdV1)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+		}
+		patchedCRDs = append(patchedCRDs, crd)
```
[the remark this answers is not in the document]
Comment (tfussell) [reply pkg/crd/renderer.go:41]: No. There are still v1beta1 CRDs in CAPZ (or at least there were a few months ago). Once those are gone, it's unlikely any more will show up, but it doesn't hurt.

The idea was that we didn't want to patch v1beta1 CRDs, but we still needed to install them for Azure AAD to work. This requirement may have changed, but as I said above, I haven't checked every CRD for every provider.

**Comment by tfussell** [comment]
> What's the difference between an upstream repository and a remote repository?

Good point. I had trouble with the naming. In my view an upstream repo is one we don't own. These all follow the CAPI provider contract so they attach infrastructure manifests to the GitHub releases. A "remote" repo would be a GS repo that defines one or more CRD, but it's not attached to the GitHub release (because we don't have a CI pipeline for that currently). I can add some comments to make this clearer. Let me know if you have an idea for better naming.

**Review by MarcelMue — APPROVED** [review]
This generally looks fine to me unless I am missing something obvious

File: pkg/crd/renderer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fiunchinho) [reply pkg/crd/renderer.go:41]: Fair enough.

**Review by fiunchinho — APPROVED** [review]

File: hack/assets.go [hunk 2]
```diff
@@ -70,3 +70,5 @@ var upstreamReleaseAssets = []crd.ReleaseAssetFileDefinition{
 		Provider: "vsphere",
 	},
 }
+
+var remoteRepositories []crd.RemoteRepositoryDefinition
```
Comment (kopiczko) [inline hack/assets.go:74]: Is this something to be filled before all of that starts to actually do anything?

**Comment by kopiczko** [comment]
Would it make sense to add some description for this in the readme?

File: hack/assets.go (hunk 2 above)
Comment (tfussell) [reply hack/assets.go:74]: Exactly. I wanted to get the structure in place so as we actually move CRDs out of apiextensions we can add them here.

**Comment by tfussell** [comment]
> Would it make sense to add some description for this in the readme?

I've updated the README now.

**Review by fiunchinho — APPROVED** [review]

**Review by kopiczko — APPROVED** [review]

**Pull request merged**